yarn-project/circuits.js/src/structs/complete_address.test.ts

```diff
@@ -4,16 +4,29 @@ import { Fr, Point } from '@aztec/foundation/fields';
 import { CompleteAddress } from './complete_address.js';
 
 describe('CompleteAddress', () => {
-  // TODO(#5834): re-enable or remove this test
-  it.skip('refuses to add an account with incorrect address for given partial address and pubkey', () => {
-    expect(() => CompleteAddress.create(AztecAddress.random(), Point.random(), Fr.random())).toThrow(
-      /cannot be derived/,
-    );
+  it('refuses to add an account with incorrect address for given partial address and pubkey', () => {
+    expect(() =>
+      CompleteAddress.create(
+        AztecAddress.random(),
+        Point.random(),
+        Point.random(),
+        Point.random(),
+        Point.random(),
+        Fr.random(),
+      ),
+    ).toThrow(/cannot be derived/);
   });
 
   it('equals returns true when 2 instances are equal', () => {
     const address1 = CompleteAddress.random();
-    const address2 = CompleteAddress.create(address1.address, address1.publicKey, address1.partialAddress);
+    const address2 = CompleteAddress.create(
+      address1.address,
```

Comment:
This will always be a pain. I still kinda would like it to not be a "CompleteAddress" because it is more address and "metadata" now I guess. Just feels too wrong that there are many things being "_address", too easy to mess up.

partial => stuff around initialization
address => what you actually use
complete => address and metadata

Reply:
I would also retire the name of CompleteAddress. Which alternative would you like?

Something like AddressPreimage might be the nice since it would also kinda communicate that it does not contain rotated keys. Might be a shit name for external devs though.

Reply:
Might make sense to tackle the renaming after key rotation is handled since that I will affect a lot where we actually use the type.